src/analyses/interval_domain.cpp

```diff
@@ -81,10 +81,13 @@ void interval_domaint::transform(
       // GOTO implies from and to are comparable
       locationt next=from;
       next++;
-      if(next==to)
-        assume(not_exprt(instruction.guard), ns);
-      else
-        assume(instruction.guard, ns);
+      if(from->get_target()!=next)     // If equal then a skip
+      {
+        if(next==to)
+          assume(not_exprt(instruction.guard), ns);
+        else
+          assume(instruction.guard, ns);
+      }
```

Comment:
Would you have a regression test to cover this?

Reply:
I don't because I believe it only occurs on transformed programs.  IIRC remove_skip removes these so it is non-trivial to create them.  I'm fixing this because both @peterschrammel and I believe that they are a case that should be handled correctly; even if it doesn't happen particularly often, see https://github.com/diffblue/cbmc/issues/1746.